### ISSUE TITLE
Handling errors

### ISSUE THREAD

**Issue by N1ark** [mokkabonna/inquirer-autocomplete-prompt#152]
I've been using the plugin, and noticed that it doesn't seem possible to properly catch errors thrown within the `source` async function to allow for a graceful exit with `this.error` if something goes wrong. No error message is printed either, and calling `this.error` inside the async function doesn't work either.

Is there something I missed?

Example code:
```ts
try {
  await inquirer
    .prompt([
      {
        type: "autocomplete",
        name: "test",
        message: "Will quit",
        source: async () => { throw Error() },
      },
    ])
    .catch(() => console.log("This is never printed"));
} finally {
  console.log("This is never printed");
}
```

**Comment by mokkabonna** [comment]
Yeah that should have been handled better. In the newer `inquirer-autocomplete-standalone` package it is handled and presented as an error if it fails.

![image](https://github.com/mokkabonna/inquirer-autocomplete-prompt/assets/230877/68223aef-5146-487c-b258-aa038418d203)

**Comment by mokkabonna** [comment]
Not sure why I never implemented that, I guess I never had the problem myself so forgot/ignored, and nobody else mentioned it either until now :)

**Comment by N1ark** [comment]
Nice ! Yeah understandable hahah, thanks for implementing it :D

**Comment by mokkabonna** [comment]
The maintained standalone package handles source errors. The legacy package is now compatibility-only for Inquirer 9, so we will not make further behavioral changes there. Closing.
